baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/safety/SafetyPropagationTransfer.java

```diff
@@ -173,9 +177,30 @@
                     .namedAnyOf("of", "copyOf"),
             MethodMatchers.staticMethod().onClass(Arrays.class.getName()).named("asList"));
 
+    private static final Matcher<ExpressionTree> OPTIONAL_FACTORIES = Matchers.anyOf(
+            MethodMatchers.staticMethod().onClass(Optional.class.getName()).namedAnyOf("of", "ofNullable"),
+            MethodMatchers.staticMethod()
+                    .onClassAny(
+                            OptionalInt.class.getName(), OptionalLong.class.getName(), OptionalDouble.class.getName())
+                    .named("of"));
+
     // These methods do not take the receiver (generally a static class) into account, only the inputs.
     private static final Matcher<ExpressionTree> RETURNS_SAFETY_COMBINATION_OF_ARGS =
-            Matchers.anyOf(STRING_FORMAT, OBJECTS_TO_STRING, IMMUTABLE_COLLECTION_FACTORY);
+            Matchers.anyOf(STRING_FORMAT, OBJECTS_TO_STRING, IMMUTABLE_COLLECTION_FACTORY, OPTIONAL_FACTORIES);
+
+    private static final Matcher<ExpressionTree> OPTIONAL_ACCESSORS = Matchers.anyOf(
+            MethodMatchers.instanceMethod()
+                    .onDescendantOf(Optional.class.getName())
+                    .namedAnyOf("get", "orElseThrow", "stream"),
```

Comment:
Do we need to handle `filter`, `orElse`, `orElseGet`?

Do we need to handle `Stream`-like `map` and `flatMap`?

Reply:
I'd like to, but I haven't included those in the initial optional support -- I'll need to implement something to determine safety of map/flatmap transformer results, and merge that with the safety of the receiver.
I want to avoid considering something like this safe:
```java
@Safe Object f(@Safe @Nullable String input, @DoNotLog password) {
    return Optional.ofNullable(input).orElseGet(() -> password);
}
```
It wouldn't be the end of the world to consider it safe rather than unknown since we don't currently suggest replacing UnsafeArg with SafeArg, but I'd like to try handling it based on all the data available first if I can.

Reply:
makes sense as a separate follow up. I think we should be able to handle filter as-is since its not widening or transforming the types.

```suggestion
                    .namedAnyOf("filter", "get", "orElseThrow", "stream"),
```

Reply:
Great idea, thanks!